watertap/examples/flowsheets/MD/MD_single_stage_continuous_recirculation.py

```diff
@@ -0,0 +1,662 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    Constraint,
+    Objective,
+    Var,
+    NonNegativeReals,
+    TransformationFactory,
+    units as pyunits,
+    check_optimal_termination,
+)
+from pyomo.network import Arc
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import (
+    propagate_state,
+)
+from idaes.models.unit_models import Heater, Separator, Mixer, Product, Feed
+from idaes.models.unit_models.separator import SplittingType
+from idaes.models.unit_models.mixer import MomentumMixingType
+from idaes.models.unit_models.heat_exchanger import (
+    HeatExchanger,
+    HeatExchangerFlowPattern,
+)
+from watertap.unit_models.mvc.components.lmtd_chen_callback import (
+    delta_temperature_chen_callback,
+)
+from watertap.unit_models.pressure_changer import Pump
+from idaes.core import UnitModelCostingBlock, FlowDirection
+import idaes.core.util.scaling as iscale
+from watertap.unit_models.MD.membrane_distillation_0D import MembraneDistillation0D
+from watertap.unit_models.MD.MD_channel_base import (
+    ConcentrationPolarizationType,
+    TemperaturePolarizationType,
+    MassTransferCoefficient,
+    PressureChangeType,
+)
+import watertap.property_models.seawater_prop_pack as props_sw
+import watertap.property_models.water_prop_pack as props_w
+from watertap.costing import WaterTAPCosting
+from watertap.costing.unit_models.pump import (
+    cost_pump,
+)
+from watertap.costing.unit_models.heater_chiller import (
+    cost_heater_chiller,
+)
+from watertap.core.util.initialization import assert_degrees_of_freedom
+
+__author__ = "Elmira Shamlou"
+
+
+def main():
+    solver = get_solver()
+    m = build()
+    set_operating_conditions(m)
+    initialize_system(m, solver=solver)
+
+    optimize_set_up(m)
+    solve(m, solver=solver)
+
+    print("\n***---Simulation results---***")
```

Comment:
Should simulated results message go after initialization/solving the square problem, then follow up here with printing "optimization results" (since results here would follow `optimize_setup`)?

Reply:
addressed